src/util/rules.ts

```diff
@@ -118,3 +118,43 @@ export function isValidCreditCardExpirationDate(
 
     return true
 }
+
+/**
+ * Checks whether the input is undefined, an empty string, or a valid VAT identification number (VATIN).
+ * @param input - The string to be validated.
+ * @returns Returns true if the input is undefined, an empty string, or a valid VATIN. Otherwise, returns a string error message indicating the validation failure.
+ */
+export function isUndefinedOrEmptyStringOrValidVATIN(
+    input: string
+): boolean | string {
+    if (input == undefined) {
+        return true
+    }
+
+    if (typeof input === 'string' && input.length === 0) {
+        return true
+    }
+
+    return isValidVATIN(input)
+}
+
+/**
+ * Checks whether the input string is a valid VAT identification number (VATIN).
+ * The pattern to match was derived from the specification of a VATIN's structure on Wikipedia:
+ * https://en.wikipedia.org/wiki/VAT_identification_number
+ * @param input - The string to be validated as a VATIN.
+ * @returns Returns true if the input is a valid VATIN, otherwise returns a string error message indicating the validation failure.
+ */
+export function isValidVATIN(input: string): boolean | string {
+    if (typeof input !== 'string' || input.length === 0) {
+        return false
+    }
+
+    const patternToMatch = /^(EU|[A-Z]{2}|EL|XI)[\dA-Z]{2,13}$/
```

Comment:
where does this come from? at least the Wikipedia article also mentions that dashes are used quite commonly, and I do not think this regex allows it

Reply:
Which article are you referring to? I could only find that Poland allows hyphens (dashes) for readability but formally their numbers only consist of digits. Of course, we could allow hyphens (dashes) and simply remove them as soon as we pass the value to the service. How about that?

Reply:
00fb73f28089e2bd0ced3fa889dac139e14d6745

Reply:
the article you reference in your docs: https://en.wikipedia.org/wiki/VAT_identification_number

below, there are some non-EU countries, some use hyphen
but I don't know if these are required